rust/gitxetcore/src/summaries/summaries_plumb.rs

```diff
@@ -316,3 +316,196 @@ pub async fn summaries_dump(config: XetConfig) -> errors::Result<()> {
     println!("{json}");
     Ok(())
 }
+
+#[cfg(test)]
```

Comment:
I'm not sure that this test format is a good idea here...  The issue is that there is no guaranteed ordering around which tests run first, especially when they are run as part of the larger process.   Also, it would be better to write out temp files to a tempfile.TempDir() path, which gets cleaned up when the object is destroyed.  

Looking at these tests, however, do you need a temporary file for most of them?  It seems like you could have the serialization function just return a Vec<u8> and then pass that to the deserialization routines, correct?

Reply:
So the issue is that I needed to have the serialization and deserialization happen on different versions of the codebase (i.e. main, this-branch, v2-branch) since each branch produces a different DB and I want to evaluate backwards and forwards compatibility. I can't run these tests automatically since I can't change the branch of this repo dynamically inside the test (thus the serialization to a permanent file). 
Note that all of the tests are `#[ignored]` so they don't run automatically.

However, I might just remove these tests since I've saved the testing code and results in an investigation doc: https://www.notion.so/xethub/Summary-Versioning-Compatibility-21f173305a9c44dea05c33ed5b176546

Reply:
Chatted offline about this and have a solid way forward for testing down the way.  Getting this in now.